apps/rig_outbound_gateway/lib/rig_outbound_gateway/kafka/sup.ex

```diff
@@ -40,10 +40,22 @@ defmodule RigOutboundGateway.Kafka.Sup do
   def init(:ok) do
     conf = config()
 
-    client_conf = [
-      auto_start_producers: true,
-      default_producer_config: []
-    ]
+    if conf.ssl == nil and conf.sasl != nil do
+      Logger.warn("SASL is enabled, but SSL is not - credentials are transmitted as cleartext.")
+    end
+
+    add_sasl_option = fn opts ->
+      if conf.sasl, do: Keyword.put(opts, :sasl, conf.sasl), else: opts
+    end
+
+    client_conf =
+      [
+        endpoints: conf.brokers,
+        auto_start_producers: true,
+        default_producer_config: [],
+        ssl: conf.ssl
```

Comment:
I guess this covers also SSL setup for producer, right?

Reply:
I've added a [test for this](https://github.com/Accenture/reactive-interaction-gateway/blob/79-kafka-ssl/apps/rig/test/kafka_test.exs#L22):
```
Rig.KafkaTest
  * test publishing using an actual connection (66.2ms)
```